exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/FindLimit0Visitor.java

```diff
@@ -46,6 +51,32 @@ public static boolean containsLimit0(RelNode rel) {
     return visitor.isContains();
   }
 
+  public static DrillRel addLimitOnTopOfLeafNodes(final DrillRel rel) {
+    final RelShuttleImpl shuttle = new RelShuttleImpl() {
+
+      private RelNode addLimitAsParent(RelNode node) {
+        final RexBuilder builder = node.getCluster().getRexBuilder();
+        final RexLiteral offset = builder.makeExactLiteral(BigDecimal.ZERO);
+        final RexLiteral fetch = builder.makeExactLiteral(BigDecimal.ZERO);
+        return new DrillLimitRel(node.getCluster(), node.getTraitSet(), node, offset, fetch);
+      }
+
+      @Override
+      public RelNode visit(TableScan scan) {
```

Comment:
You also need override visitValues, since Value could be leaf operator as well, in addition to TableScan.